components/ai_chat/resources/page/components/filter_menu/filter_menu.tsx

```diff
@@ -136,6 +139,36 @@ export default function FilterMenu<T>(props: Props<T>) {
     }
   }, [props.isOpen, props.query])
 
+  React.useEffect(() => {
+    if (!props.isOpen) {
+      props.onSelectedItemChanged?.(null)
+      return
+    }
+
+    const handler = () => {
+      const focused =
+        ref.current?.querySelector<HTMLElement>(':focus')
+        ?? ref.current?.querySelector<HTMLElement>('leo-menu-item')
+      if (!focused) {
+        props.onSelectedItemChanged?.(null)
+        return
+      }
+
+      const item = domToItem.current.get(focused)
+      if (!item) {
+        props.onSelectedItemChanged?.(null)
+        return
+      }
+
+      props.onSelectedItemChanged?.(item)
+    }
+    handler()
+    ref.current?.addEventListener('focusin', handler)
+    return () => {
+      ref.current?.removeEventListener('focusout', handler)
+    }
+  }, [props.isOpen, props.onSelectedItemChanged, filtered])
```

Comment:
Would be nice for this block to have at least one comment explaining what it's for.